keps/sig-node/4328-node-label-manager/README.md

```diff
@@ -0,0 +1,442 @@
+
+Inspect the nodeAffinity.RequiredDuringSchedulingRequiredDuringExecution configuration
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: NodeLabelsManagerEviction
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [x] Other (treat as last resort)
+  - Details: pod with `RequiredDuringSchedulingRequiredDuringExecution` must meet affinity rule during both the scheduling phase and the execution phase.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: 
+    - pod_deletions_total(counts the number of Pods deleted since its start)
+    - pod_deletion_duration_seconds(tracks the latency between the time when a node affinity effect has been activated for the Pod and its deletion.)
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
+
+### Dependencies
+
+N/A
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+NodeAffinity have one new field `RequiredDuringSchedulingRequiredDuringExecution`, increasing their size slightly.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+A new field `RequiredDuringSchedulingRequiredDuringExecution` will be added to NodeAffinity. It uses the existing api type `NodeSelector`.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+It increasing the size of `NodeAffinity` a little bit.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+If `RequiredDuringSchedulingRequiredDuringExecution` field is defined, this feature will slightly increase the pod scheduling time.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+The feature will not work.
+
+###### What are other known failure modes?
+
+N/A
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+- 2023-11-06: Initial draft KEP
+
+## Drawbacks
+
+N/A
+
+## Alternatives
+
+- taints
```

Comment:
Both of these alternatives should be expanded upon.  This is my main questions with this KEP is why does the controller have to be brand-new/does it have to be in k/k.